Index: socialcalctableeditor.js
===================================================================
--- socialcalctableeditor.js	(revision 65)
+++ socialcalctableeditor.js	(working copy)
@@ -186,6 +186,9 @@
 
    this.range = {hasrange: false};
 
+   // Relative offset of editor when it's embedded inside relative-positioned ancestors
+   this.relativeoffset = {left:0, top:0};
+
    }
 
 // Methods:
@@ -829,8 +832,8 @@
    sizedisplay.style.width = "auto";
    sizedisplay.style.position = "absolute";
    sizedisplay.style.zIndex = 100;
-   sizedisplay.style.top = (editor.headposition.top+0)+"px";
-   sizedisplay.style.left = editor.colpositions[result.coltoresize]+"px";
+   sizedisplay.style.top = (editor.headposition.top-editor.relativeoffset.top)+"px";
+   sizedisplay.style.left = (editor.colpositions[result.coltoresize]-editor.relativeoffset.left)+"px";
    sizedisplay.innerHTML = '<table cellpadding="0" cellspacing="0"><tr><td style="height:100px;'+
       'border:1px dashed black;background-color:white;width:' +
       (editor.context.colwidth[mouseinfo.mouseresizecolnum]-2) + 'px;">&nbsp;</td>'+
@@ -1843,6 +1846,9 @@
    editor.lastnonscrollingcol = editor.context.colpanes.length-1 > 0 ?
          editor.context.colpanes[editor.context.colpanes.length-2].last : 0;
 
+   // Relative offset
+   editor.relativeoffset = SocialCalc.GetRelativeOffset(editor.toplevel);
+
    // Now do the table controls
 
    editor.verticaltablecontrol.ComputeTableControlPositions();
@@ -2479,8 +2485,8 @@
       cell=SocialCalc.GetEditorCellElement(editor, editor.ecell.row, editor.ecell.col);
       if (cell) {
          position = SocialCalc.GetElementPosition(cell.element);
-         inputecho.main.style.left = (position.left-1)+"px";
-         inputecho.main.style.top = (position.top-1)+"px";
+         inputecho.main.style.left = (position.left-1-editor.relativeoffset.left)+"px";
+         inputecho.main.style.top = (position.top-1-editor.relativeoffset.top)+"px";
          }
       inputecho.main.style.display = "block";
       if (inputecho.interval) window.clearInterval(inputecho.interval); // just in case
@@ -2847,26 +2853,27 @@
 SocialCalc.ComputeTableControlPositions = function(control) {
 
    var editor = control.editor;
+   var offset = control.editor.relativeoffset;
 
    if (!editor.gridposition || !editor.headposition) throw("Can't compute table control positions before editor positions");
 
    if (control.vertical) {
-      control.controlborder = editor.gridposition.left+editor.tablewidth; // border=left position
-      control.endcapstart = editor.gridposition.top; // start=top position
-      control.panesliderstart = editor.firstscrollingrowtop-control.sliderthickness;
-      control.lessbuttonstart = editor.firstscrollingrowtop-1;
-      control.morebuttonstart = editor.gridposition.top+editor.tableheight-control.buttonthickness;
-      control.scrollareastart = editor.firstscrollingrowtop-1+control.buttonthickness;
+      control.controlborder = editor.gridposition.left+editor.tablewidth-offset.left; // border=left position
+      control.endcapstart = editor.gridposition.top-offset.top; // start=top position
+      control.panesliderstart = editor.firstscrollingrowtop-control.sliderthickness-offset.top;
+      control.lessbuttonstart = editor.firstscrollingrowtop-1-offset.top;
+      control.morebuttonstart = editor.gridposition.top+editor.tableheight-control.buttonthickness-offset.top;
+      control.scrollareastart = editor.firstscrollingrowtop-1+control.buttonthickness-offset.top;
       control.scrollareaend = control.morebuttonstart-1;
       control.scrollareasize = control.scrollareaend-control.scrollareastart+1;
       }
    else {
-      control.controlborder = editor.gridposition.top+editor.tableheight; // border=top position
-      control.endcapstart = editor.gridposition.left; // start=left position
-      control.panesliderstart = editor.firstscrollingcolleft-control.sliderthickness;
-      control.lessbuttonstart = editor.firstscrollingcolleft-1;
-      control.morebuttonstart = editor.gridposition.left+editor.tablewidth-control.buttonthickness;
-      control.scrollareastart = editor.firstscrollingcolleft-1+control.buttonthickness;
+      control.controlborder = editor.gridposition.top+editor.tableheight-offset.top; // border=top position
+      control.endcapstart = editor.gridposition.left-offset.left; // start=left position
+      control.panesliderstart = editor.firstscrollingcolleft-control.sliderthickness-offset.left;
+      control.lessbuttonstart = editor.firstscrollingcolleft-1-offset.left;
+      control.morebuttonstart = editor.gridposition.left+editor.tablewidth-control.buttonthickness-offset.left;
+      control.scrollareastart = editor.firstscrollingcolleft-1+control.buttonthickness-offset.left;
       control.scrollareaend = control.morebuttonstart-1;
       control.scrollareasize = control.scrollareaend-control.scrollareastart+1;
       }
@@ -2898,13 +2905,13 @@
 
    if (dobj.vertical) {
       row = SocialCalc.Lookup(draginfo.clientY+dobj.functionobj.control.sliderthickness, editor.rowpositions);
-      draginfo.trackingline.style.top = (editor.rowpositions[row] || editor.headposition.top)+"px";
-      draginfo.trackingline.style.left = editor.headposition.left+"px";
+      draginfo.trackingline.style.top = ((editor.rowpositions[row] || editor.headposition.top)-editor.relativeoffset.top)+"px";
+      draginfo.trackingline.style.left = (editor.headposition.left-editor.relativeoffset.left)+"px";
       }
    else {
       col = SocialCalc.Lookup(draginfo.clientX+dobj.functionobj.control.sliderthickness, editor.colpositions);
-      draginfo.trackingline.style.top = editor.headposition.top+"px";
-      draginfo.trackingline.style.left = (editor.colpositions[col] || editor.headposition.left)+"px";
+      draginfo.trackingline.style.top = (editor.headposition.top-editor.relativeoffset.top)+"px";
+      draginfo.trackingline.style.left = ((editor.colpositions[col] || editor.headposition.left)-editor.relativeoffset.left)+"px";
       }
 
    editor.griddiv.appendChild(draginfo.trackingline);
@@ -2925,20 +2932,20 @@
    if (dobj.vertical) {
       max = control.morebuttonstart - control.minscrollingpanesize - draginfo.offsetY; // restrict movement
       if (draginfo.clientY > max) draginfo.clientY = max;
-      min = editor.headposition.top - sliderthickness - draginfo.offsetY;
+      min = editor.headposition.top - sliderthickness - draginfo.offsetY - editor.relativeoffset.top;
       if (draginfo.clientY < min) draginfo.clientY = min;
 
       row = SocialCalc.Lookup(draginfo.clientY+sliderthickness, editor.rowpositions);
-      draginfo.trackingline.style.top = (editor.rowpositions[row] || editor.headposition.top)+"px";
+      draginfo.trackingline.style.top = ((editor.rowpositions[row] || editor.headposition.top)-editor.relativeoffset.top)+"px";
       }
    else {
       max = control.morebuttonstart - control.minscrollingpanesize - draginfo.offsetX;
       if (draginfo.clientX > max) draginfo.clientX = max;
-      min = editor.headposition.left - sliderthickness - draginfo.offsetX;
+      min = editor.headposition.left - sliderthickness - draginfo.offsetX - editor.relativeoffset.left;
       if (draginfo.clientX < min) draginfo.clientX = min;
 
       col = SocialCalc.Lookup(draginfo.clientX+sliderthickness, editor.colpositions);
-      draginfo.trackingline.style.left = (editor.colpositions[col] || editor.headposition.left)+"px";
+      draginfo.trackingline.style.left = ((editor.colpositions[col] || editor.headposition.left)-editor.relativeoffset.left)+"px";
       }
 
    SocialCalc.DragFunctionPosition(event, draginfo, dobj);
@@ -2959,7 +2966,7 @@
    if (dobj.vertical) {
       max = control.morebuttonstart - control.minscrollingpanesize - draginfo.offsetY; // restrict movement
       if (draginfo.clientY > max) draginfo.clientY = max;
-      min = editor.headposition.top - sliderthickness - draginfo.offsetY;
+      min = editor.headposition.top - sliderthickness - draginfo.offsetY - editor.relativeoffset.top;
       if (draginfo.clientY < min) draginfo.clientY = min;
 
       row = SocialCalc.Lookup(draginfo.clientY+sliderthickness, editor.rowpositions);
@@ -2979,7 +2986,7 @@
    else {
       max = control.morebuttonstart - control.minscrollingpanesize - draginfo.offsetX;
       if (draginfo.clientX > max) draginfo.clientX = max;
-      min = editor.headposition.left - sliderthickness - draginfo.offsetX;
+      min = editor.headposition.left - sliderthickness - draginfo.offsetX - editor.relativeoffset.left;
       if (draginfo.clientX < min) draginfo.clientX = min;
 
       col = SocialCalc.Lookup(draginfo.clientX+sliderthickness, editor.colpositions);
@@ -3035,13 +3042,13 @@
    SocialCalc.setStyles(draginfo.thumbstatus, scc.TCTDFSthumbstatusStyle);
 
    if (dobj.vertical) {
-      draginfo.thumbstatus.style.top = (draginfo.clientY+scc.TCTDFStopOffsetv)+"px";
+      draginfo.thumbstatus.style.top = (draginfo.clientY+scc.TCTDFStopOffsetv-editor.relativeoffset.top)+"px";
       draginfo.thumbstatus.style.left = (control.controlborder+scc.TCTDFSleftOffsetv)+"px";
       draginfo.thumbstatus.innerHTML = scc.s_TCTDFthumbstatusPrefixv+editor.firstscrollingrow;
       }
    else {
       draginfo.thumbstatus.style.top = (control.controlborder+scc.TCTDFStopOffseth)+"px";
-      draginfo.thumbstatus.style.left = (draginfo.clientX+scc.TCTDFSleftOffseth)+"px";
+      draginfo.thumbstatus.style.left = (draginfo.clientX+scc.TCTDFSleftOffseth-editor.relativeoffset.left)+"px";
       draginfo.thumbstatus.innerHTML = scc.s_TCTDFthumbstatusPrefixh+SocialCalc.rcColname(editor.firstscrollingcol);
       }
 
@@ -3067,7 +3074,7 @@
          draginfo.clientY = control.scrollareaend - draginfo.offsetY - control.thumbthickness + 2;
       if (draginfo.clientY < control.scrollareastart - draginfo.offsetY - 1)
          draginfo.clientY = control.scrollareastart - draginfo.offsetY - 1;
-      draginfo.thumbstatus.style.top = draginfo.clientY+"px";
+      draginfo.thumbstatus.style.top = (draginfo.clientY-editor.relativeoffset.top)+"px";
 
       first =
          ((draginfo.clientY+draginfo.offsetY-control.scrollareastart+1)/(control.scrollareasize-control.thumbthickness))
@@ -3084,7 +3091,7 @@
          draginfo.clientX = control.scrollareaend - draginfo.offsetX - control.thumbthickness + 2;
       if (draginfo.clientX < control.scrollareastart - draginfo.offsetX - 1)
          draginfo.clientX = control.scrollareastart - draginfo.offsetX - 1;
-      draginfo.thumbstatus.style.left = draginfo.clientX+"px";
+      draginfo.thumbstatus.style.left = (draginfo.clientX-editor.relativeoffset.left)+"px";
 
       first =
          ((draginfo.clientX+draginfo.offsetX-control.scrollareastart+1)/(control.scrollareasize-control.thumbthickness))
Index: socialcalc-3.js
===================================================================
--- socialcalc-3.js	(revision 65)
+++ socialcalc-3.js	(working copy)
@@ -3588,6 +3589,47 @@
    }
 
 //
+// GetComputedStyle(element, style) - returns computed style value
+//
+// http://blog.stchur.com/2006/06/21/css-computed-style/
+//
+
+SocialCalc.GetComputedStyle = function (element, style) {
+
+   var computedStyle;
+   if (typeof element.currentStyle != 'undefined') { // IE
+      computedStyle = element.currentStyle;
+      }
+   else {
+      computedStyle = document.defaultView.getComputedStyle(element, null);
+      }
+   return computedStyle[style];
+
+   }
+
+//
+// GetRelativeOffset(element) - returns relative offset of element
+// The relative offset is the offset of the first relative-positioned ancestor of the element.
+//
+
+SocialCalc.GetRelativeOffset = function (element) {
+
+   var e = element;
+   var offsetLeft = 0, offsetTop = 0;
+   while (e) {
+      if (SocialCalc.GetComputedStyle(e, 'position')=='relative') {
+         offset = SocialCalc.GetElementPosition(e);
+         offsetLeft += offset.left;
+         offsetTop  += offset.top;
+         break;
+         }
+      e = e.parentNode;
+      }
+   return {left:offsetLeft, top:offsetTop};
+
+   }
+
+//
 // LookupElement(element, array) - returns array element which is an object with "element" of element
 //
 
