Index: socialcalc-3.js
===================================================================
--- socialcalc-3.js	(revision 65)
+++ socialcalc-3.js	(working copy)
@@ -3139,7 +3139,7 @@
       for (colnum=context.colpanes[colpane].first; colnum<=context.colpanes[colpane].last; colnum++) {
          newcol=document.createElement("col");
          t = context.colwidth[colnum];
-         if (t) newcol.width=t;
+         if (t > 0) newcol.width=t;
          result.appendChild(newcol);
          }
       if (colpane<context.colpanes.length-1) {
@@ -3169,7 +3169,7 @@
       for (colnum=context.colpanes[colpane].first; colnum<=context.colpanes[colpane].last; colnum++) {
          newcell=document.createElement("td");
          t = context.colwidth[colnum];
-         if (t) newcell.width=t;
+         if (t > 0) newcell.width=t;
          newcell.height="1";
          result.appendChild(newcell);
          }
