Index: socialcalc-3.js
===================================================================
--- socialcalc-3.js	(revision 65)
+++ socialcalc-3.js	(working copy)
@@ -3522,14 +3523,18 @@
 
 SocialCalc.setStyles = function (element, cssText) {
 
-   var parts, part, namevalue;
+   var parts, part, pos, name, value;
 
    if (!cssText) return;
 
    parts = cssText.split(";");
    for (part=0; part<parts.length; part++) {
-      namevalue = parts[part].split(":");
-      if (namevalue[0]) element.style[namevalue[0]] = namevalue[1];
+      pos = parts[part].indexOf(":");
+      if (pos != -1) {
+         name = parts[part].substring(0,pos);
+         value = parts[part].substring(pos+1);
+         if (name && value) element.style[name] = value;
+         }
       }
 
    }
