Index: socialcalc-3.js
===================================================================
--- socialcalc-3.js	(revision 65)
+++ socialcalc-3.js	(working copy)
@@ -388,7 +388,7 @@
                      attribs.lastrow=parts[j++]-0;
                      break;
                   case "w":
-                     attribs.defaultcolwidth=parts[j++]-0;
+                     attribs.defaultcolwidth=parts[j++]+"";
                      break;
                   case "h":
                      attribs.defaultrowheight=parts[j++]-0;
