Index: socialcalcspreadsheetcontrol.js
===================================================================
--- socialcalcspreadsheetcontrol.js	(revision 65)
+++ socialcalcspreadsheetcontrol.js	(working copy)
@@ -199,33 +199,33 @@
    this.tabnums.edit = this.tabs.length;
    this.tabs.push({name: "edit", text: "Edit", html:
       ' <div id="%id.edittools" style="padding:10px 0px 0px 0px;">'+
-'&nbsp;<img id="%id.button_undo" src="images/sc-undo.gif" style="vertical-align:bottom;">'+
-' <img id="%id.button_redo" src="images/sc-redo.gif" style="vertical-align:bottom;">'+
-' &nbsp;<img src="images/sc-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
-'<img id="%id.button_copy" src="images/sc-copy.gif" style="vertical-align:bottom;">'+
-' <img id="%id.button_cut" src="images/sc-cut.gif" style="vertical-align:bottom;">'+
-' <img id="%id.button_paste" src="images/sc-paste.gif" style="vertical-align:bottom;">'+
-' &nbsp;<img src="images/sc-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
-'<img id="%id.button_delete" src="images/sc-delete.gif" style="vertical-align:bottom;">'+
-' <img id="%id.button_pasteformats" src="images/sc-pasteformats.gif" style="vertical-align:bottom;">'+
-' &nbsp;<img src="images/sc-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
-'<img id="%id.button_filldown" src="images/sc-filldown.gif" style="vertical-align:bottom;">'+
-' <img id="%id.button_fillright" src="images/sc-fillright.gif" style="vertical-align:bottom;">'+
-' &nbsp;<img src="images/sc-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
-'<img id="%id.button_alignleft" src="images/sc-alignleft.gif" style="vertical-align:bottom;">'+
-' <img id="%id.button_aligncenter" src="images/sc-aligncenter.gif" style="vertical-align:bottom;">'+
-' <img id="%id.button_alignright" src="images/sc-alignright.gif" style="vertical-align:bottom;">'+
-' &nbsp;<img src="images/sc-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
-'<img id="%id.button_borderon" src="images/sc-borderson.gif" style="vertical-align:bottom;"> '+
-' <img id="%id.button_borderoff" src="images/sc-bordersoff.gif" style="vertical-align:bottom;"> '+
-' &nbsp;<img src="images/sc-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
-'<img id="%id.button_merge" src="images/sc-merge.gif" style="vertical-align:bottom;"> '+
-' <img id="%id.button_unmerge" src="images/sc-unmerge.gif" style="vertical-align:bottom;"> '+
-' &nbsp;<img src="images/sc-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
-'<img id="%id.button_insertrow" src="images/sc-insertrow.gif" style="vertical-align:bottom;"> '+
-' <img id="%id.button_insertcol" src="images/sc-insertcol.gif" style="vertical-align:bottom;"> '+
-'&nbsp; <img id="%id.button_deleterow" src="images/sc-deleterow.gif" style="vertical-align:bottom;"> '+
-' <img id="%id.button_deletecol" src="images/sc-deletecol.gif" style="vertical-align:bottom;"> '+
+'&nbsp;<img id="%id.button_undo" src="'+SocialCalc.Constants.defaultImagePrefix+'-undo.gif" style="vertical-align:bottom;">'+
+' <img id="%id.button_redo" src="'+SocialCalc.Constants.defaultImagePrefix+'-redo.gif" style="vertical-align:bottom;">'+
+' &nbsp;<img src="'+SocialCalc.Constants.defaultImagePrefix+'-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
+'<img id="%id.button_copy" src="'+SocialCalc.Constants.defaultImagePrefix+'-copy.gif" style="vertical-align:bottom;">'+
+' <img id="%id.button_cut" src="'+SocialCalc.Constants.defaultImagePrefix+'-cut.gif" style="vertical-align:bottom;">'+
+' <img id="%id.button_paste" src="'+SocialCalc.Constants.defaultImagePrefix+'-paste.gif" style="vertical-align:bottom;">'+
+' &nbsp;<img src="'+SocialCalc.Constants.defaultImagePrefix+'-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
+'<img id="%id.button_delete" src="'+SocialCalc.Constants.defaultImagePrefix+'delete.gif" style="vertical-align:bottom;">'+
+' <img id="%id.button_pasteformats" src="'+SocialCalc.Constants.defaultImagePrefix+'-pasteformats.gif" style="vertical-align:bottom;">'+
+' &nbsp;<img src="'+SocialCalc.Constants.defaultImagePrefix+'-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
+'<img id="%id.button_filldown" src="'+SocialCalc.Constants.defaultImagePrefix+'-filldown.gif" style="vertical-align:bottom;">'+
+' <img id="%id.button_fillright" src="'+SocialCalc.Constants.defaultImagePrefix+'-fillright.gif" style="vertical-align:bottom;">'+
+' &nbsp;<img src="'+SocialCalc.Constants.defaultImagePrefix+'-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
+'<img id="%id.button_alignleft" src="'+SocialCalc.Constants.defaultImagePrefix+'-alignleft.gif" style="vertical-align:bottom;">'+
+' <img id="%id.button_aligncenter" src="'+SocialCalc.Constants.defaultImagePrefix+'-aligncenter.gif" style="vertical-align:bottom;">'+
+' <img id="%id.button_alignright" src="'+SocialCalc.Constants.defaultImagePrefix+'-alignright.gif" style="vertical-align:bottom;">'+
+' &nbsp;<img src="'+SocialCalc.Constants.defaultImagePrefix+'-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
+'<img id="%id.button_borderon" src="'+SocialCalc.Constants.defaultImagePrefix+'-borderson.gif" style="vertical-align:bottom;"> '+
+' <img id="%id.button_borderoff" src="'+SocialCalc.Constants.defaultImagePrefix+'-bordersoff.gif" style="vertical-align:bottom;"> '+
+' &nbsp;<img src="'+SocialCalc.Constants.defaultImagePrefix+'-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
+'<img id="%id.button_merge" src="'+SocialCalc.Constants.defaultImagePrefix+'-merge.gif" style="vertical-align:bottom;"> '+
+' <img id="%id.button_unmerge" src="'+SocialCalc.Constants.defaultImagePrefix+'-unmerge.gif" style="vertical-align:bottom;"> '+
+' &nbsp;<img src="'+SocialCalc.Constants.defaultImagePrefix+'-divider1.gif" style="vertical-align:bottom;">&nbsp; '+
+'<img id="%id.button_insertrow" src="'+SocialCalc.Constants.defaultImagePrefix+'-insertrow.gif" style="vertical-align:bottom;"> '+
+' <img id="%id.button_insertcol" src="'+SocialCalc.Constants.defaultImagePrefix+'-insertcol.gif" style="vertical-align:bottom;"> '+
+'&nbsp; <img id="%id.button_deleterow" src="'+SocialCalc.Constants.defaultImagePrefix+'-deleterow.gif" style="vertical-align:bottom;"> '+
+' <img id="%id.button_deletecol" src="'+SocialCalc.Constants.defaultImagePrefix+'-deletecol.gif" style="vertical-align:bottom;"> '+
       ' </div>',
       oncreate: null, //function(spreadsheet, viewobject) {SocialCalc.DoCmd(null, "fill-rowcolstuff");},
       onclick: null});
